llvm/lib/Transforms/InstCombine/InstCombineSimplifyDemanded.cpp

```diff
@@ -966,6 +966,48 @@ Value *InstCombinerImpl::SimplifyDemandedUseBits(Value *V, APInt DemandedMask,
                 I, 1, (DemandedMask & ~LHSKnown.Zero).zextOrTrunc(MaskWidth)))
           return I;
 
+        // Combine:
+        // (ptrmask (getelementptr i8, ptr p, imm i), imm mask)
+        //   -> (ptrmask (getelementptr i8, ptr p, imm (i & mask)), imm mask)
+        // where only the low bits known to be zero in the pointer are changed
+        Value *InnerPtr;
+        uint64_t GEPIndex;
+        uint64_t PtrMaskImmediate;
+        if (match(I, m_Intrinsic<Intrinsic::ptrmask>(
+                         m_GEP(m_Value(InnerPtr), m_ConstantInt(GEPIndex)),
```

Comment:
Instead of matching `GEP` then checking its `i8` think you can just use `m_PtrAdd`.

Reply:
There probably is a more idomatic way to do this - in particular matching i8 is less good than matching arbitrary integer with a scaling factor - though m_PtrAdd doesn't appear to exist at present

Reply:
m_PtrAdd has been added since this PR was created -- need to rebase.

Reply:
So it has. Cleans up the patch slightly, thanks!